src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs

```diff
@@ -76,9 +76,10 @@ private void Load(bool reload)
             }
             else
             {
-                static Stream OpenRead(IFileInfo fileInfo)
+                static Stream OpenRead(IFileInfo fileInfo, IFileProvider provider)
                 {
-                    if (fileInfo.PhysicalPath != null)
+                    if (fileInfo.PhysicalPath != null
+                        && provider is PhysicalFileProvider && !provider.GetType().IsSubclassOf(typeof(PhysicalFileProvider)))
```

Comment:
Couldn't this just be

```suggestion
                        && provider?.GetType() == typeof(PhysicalFileProvider))
```

Also - would be great to add a comment here why we are doing this.

Reply:
Can we do this check once in the constructor